src/chimera/interfaces/weatherstation.py

```diff
@@ -21,100 +21,125 @@
+    __accepted_temperature_units__ = [
+        units.Celsius,
+        units.Kelvin,
+        units.imperial.deg_F
+    ]
+
+    """
+    Speed units accepted by the interface.
+    """
+    __accepted_speed_units__ = [
+        units.meter / units.second,
+        units.kilometer / units.hour,
+        units.imperial.mile / units.hour,
+        units.imperial.foot / units.second,
+    ]
+
+    """
+    Direction units accepted by the interface.
+    """
+    __accepted_direction_unit__ = [
+        units.degree,
+        units.radian
+    ]
+
+    """
+    Pressure units accepted by the interface.
+    """
+    __accepted_pressures_unit__ = [
+        units.cds.mmHg,
+        units.bar,
+        units.cds.atm,
+        units.Pa]
+
+    def humidity(self, unit_out=units.pct):
+        """
+        Returns the 100% relative humidity (Default: Percentage).
+        :param unit: Unit in which the instrument should return the humidity.
+        :return: the humidity.
+        """
         pass
 
-    @event
-    def humidityChange(self, humidity, unit, delta):
+    def temperature(self, unit_out=units.Celsius):
+        """
+        Returns the temperature in the chosen unit (Default: Celsius).
+        :param unit:  Unit in which the instrument should return the temperature.
+        :return: the temperature.
+        """
         pass
 
-    @event
-    def temperatureChange(self, temperature, unit, delta):
+    def wind_speed(self, unit_out=units.meter/units.second):
+        """
+        Returns the wind speed in the chosen unit (Default: meters per second).
+        :param unit:  Unit in which the instrument should return the wind speed.
+        :return: the wind speed.
+        """
         pass
 
-    @event
-    def windChange(self, wind, unit, delta):
+    def wind_direction(self, unit_out=units.degree):
+        """
+        Returns the wind direction in the chosen unit (Default: Degrees).
+        :param unit:  Unit in which the instrument should return the wind direction.
+        :return: the wind direction.
+        """
         pass
 
-    @event
-    def dewPointChange(self, dewPoint, unit, delta):
+    def dew_point(self, unit_out=units.Celsius):
+        """
+        Returns the dew point temperature in the chosen unit (Default: Celsius).
+        :param unit:  Unit in which the instrument should return the dew point.
+        :return: the dew point temperature.
+        """
         pass
 
-    @event
-    def pressureChange(self, pressure, unit, delta):
+    def pressure(self, unit_out=units.Pa):
+        """
+        Returns the pressure in the chosen unit (Default: units.Pa).
+        :param unit:  Unit in which the instrument should return the pressure.
+        :return: the pressure.
+        """
         pass
 
-    @event
-    def rainChange(self, rain, unit, delta):
-        pass
+    def rain(self, unit_out=units.liter/units.hour):
+        """
+        Returns the precipitation rate in the chosen unit (Default: mm/H).
```

Comment:
not sure in what direction the sentence goes.

just to be clear:

mm/hour or inches/hour is right

volume/time is wrong

On Tue, Aug 4, 2015 at 12:30 PM, William Schoenell <notifications@github.com
> wrote:

> In src/chimera/interfaces/weatherstation.py
> <https://github.com/astroufsc/chimera/pull/96#discussion_r36202165>:
>
> >          pass
> >
> > -    @event
> > -    def rainChange(self, rain, unit, delta):
> > -        pass
> > +    def rain(self, unit_out=units.liter/units.hour):
> > +        """
> > +        Returns the precipitation rate in the chosen unit (Default: mm/H).
>
> Here is still wrong the doc in respect to the def. Precipitation rate unit
> can be mm/hour or inches/hour and not volume/hour.
>
> http://web.ead.anl.gov/resrad/datacoll/water.htm
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/astroufsc/chimera/pull/96/files#r36202165>.
>